Python/ceval.c

```diff
@@ -4678,43 +4670,48 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             DISPATCH();
         }
 
-        TARGET(CALL_FUNCTION_ADAPTIVE) {
+        TARGET(CALL_NO_KW_ADAPTIVE) {
             SpecializedCacheEntry *cache = GET_CACHE();
-            nargs = cache->adaptive.original_oparg;
+            oparg = cache->adaptive.original_oparg;
             if (cache->adaptive.counter == 0) {
                 next_instr--;
-                if (_Py_Specialize_CallFunction(
+                int nargs = oparg+extra_args;
```

Comment:
Don't you need to set `extra_args = 0;` after this line too?

Side note: the `extra_args` and `post_args` contracts may be human-error prone for other contributors, since people need to remember to reset them at the end of every call.

Reply:
We want to leave `extra_args` untouched because we are about to re-execute this instruction (or whatever `_Py_Specialize_CallNoKw` turns it into).